### PR TITLE
Added ISO keymap support (based on CH layout) for IDOBAO ID80

### PR DESCRIPTION
**Pull request by phi-0**
## Description

So far only ANSI keymaps / layouts were available for the IDOBAO ID80. With this small addition of the 'iso_ch' keymap for keyboard 'id80' according support was implemented. Specifically, files id80.h, info.json as well as keymap.c were adjusted to support the additional 2 keys part of the 75% ISO layout (as compared to the 75% ANSI layout). The implementation was tested with a new IDOBAO ID80 in CH ISO layout and no issues were found so far.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* None (addition of keymap / layout

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/id80/id80.h [hunk 1]
```diff
@@ -17,6 +18,7 @@
 
 #include "quantum.h"
 
+
```
Comment (fauxpark) [inline keyboards/id80/id80.h:21]:
```suggestion
```

File: keyboards/id80/info.json [hunk 2]
```diff
@@ -8,6 +8,10 @@
         "LAYOUT": {
             "key_count": 80,
```
Comment (fauxpark) [inline keyboards/id80/info.json:9]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/id80/keymaps/iso_ch/keymap.c [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_iso(
+        KC_ESC,    KC_F1,   KC_F2,   KC_F3,   KC_F4,     KC_F5,   KC_F6,   KC_F7,   KC_F8,     KC_F9,   KC_F10,  KC_F11,  KC_F12,     MO(1),     KC_DEL,
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,            KC_HOME,
+        KC_TAB,      KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_ENT,         KC_END,
+        KC_CAPS,       KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_NUHS, KC_ENT,
```
Comment (noroadsleft) [inline keyboards/id80/keymaps/iso_ch/keymap.c:24]: Does the Enter key on Line 23 actually do anything? If you flash this, and then trigger the ANSI backslash location (with a paper clip, wire, etc.), does it activate?

Comment (phi-0) [reply keyboards/id80/keymaps/iso_ch/keymap.c:24]: No it does not do anything, to my understanding this is only done for the keymap representation. 
There actually only is 1 switch across the two rows on the ISO version of the ID80 board, no separate switch for the ANSI backslash location. My understanding was that this is how I have to specify the ISO enter key (spanning two rows) in the keymap.

Comment (Erovia) [reply keyboards/id80/keymaps/iso_ch/keymap.c:24]: In this case, it'd make more sense to just drop it from the layout macro.

I'll add the suggestions that should be working, but test it on hardware.

File: keyboards/id80/id80.h [hunk 4]
```diff
@@ -35,3 +36,23 @@
     { K07,   K17,   K27,   K37,   K47,   K57,   K67,   K77,   K87,   K97,   KA7   }, \
     { K08,   K18,   K28,   K38,   K48,   K58,   K68,   K78,   K88,   K98,   KC_NO }, \
 }
+
+
+#define LAYOUT_iso( \
+    K50,  K51, K52, K53, K54,  K55, K56, K57, K58,  KA7, KA5, KA4, KA3,   KA6,  KA2,  \
+    K40, K41, K42, K43, K44, K45, K46, K47, K48, K98, K97, K95, K94, K96,       K92,  \
+    K30,   K31, K32, K33, K34, K35, K36, K37, K38, K88, K87, K85, K84, K83,     K82,  \
```
Comment (Erovia) [inline keyboards/id80/id80.h:44]:
```suggestion
    K30,   K31, K32, K33, K34, K35, K36, K37, K38, K88, K87, K85, K84,          K82,  \
```

File: keyboards/id80/id80.h [hunk 5]
```diff
@@ -35,3 +36,23 @@
...
+    K20,    K21, K22, K23, K24, K25, K26, K27, K28, K78, K77, K75, K74, K73,          \
+    K10,    K11, K12, K13, K14, K15, K16, K17, K18, K68, K67, K65, K64,      K63,     \
+    K00,  K01,  K02,              K06,                K08,   K07,     K05, K04, K03   \
+) { \
+    { K00,   K10,   K20,   K30,   K40,   K50,   KC_NO, KC_NO, KC_NO, KC_NO, KC_NO }, \
+    { K01,   K11, K21,   K31,   K41,   K51,   KC_NO, KC_NO, KC_NO, KC_NO, KC_NO }, \
+    { K02,   K12,   K22,   K32,   K42,   K52,   KC_NO, KC_NO, K82,   K92,   KA2   }, \
+    { K03,   K13,   K23,   K33,   K43,   K53,   K63,   KC_NO, K83,   KC_NO, KA3   }, \
```
Comment (Erovia) [inline keyboards/id80/id80.h:52]:
```suggestion
    { K03,   K13,   K23,   K33,   K43,   K53,   K63,   KC_NO, KC_NO, KC_NO, KA3   }, \
```

File: keyboards/id80/keymaps/iso_ch/keymap.c [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_iso(
+        KC_ESC,    KC_F1,   KC_F2,   KC_F3,   KC_F4,     KC_F5,   KC_F6,   KC_F7,   KC_F8,     KC_F9,   KC_F10,  KC_F11,  KC_F12,     MO(1),     KC_DEL,
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,            KC_HOME,
+        KC_TAB,      KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_ENT,         KC_END,
```
Comment (Erovia) [inline keyboards/id80/keymaps/iso_ch/keymap.c:23]:
```suggestion
        KC_TAB,      KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC,                 KC_END,
```

File: keyboards/id80/keymaps/iso_ch/keymap.c [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        KC_CAPS,       KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_NUHS, KC_ENT,
+        KC_LSFT,   KC_NUBS,    KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,          KC_UP,
+        KC_LCTL,   KC_LGUI,   KC_LALT,                       KC_SPC,                              KC_RALT,     KC_RCTL,        KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+    [1] = LAYOUT_iso(
+        RESET,     _______, _______, _______, _______,   _______, _______, _______, _______,   _______, _______, _______, _______,    _______,   KC_MUTE,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,            _______,
+        _______,     RGB_TOG, _______, RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, _______, _______, _______, _______,        _______,
```
Comment (Erovia) [inline keyboards/id80/keymaps/iso_ch/keymap.c:31]:
```suggestion
        _______,     RGB_TOG, _______, RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, _______, _______, _______,                 _______,
```

File: keyboards/id80/readme.md [hunk 8]
```diff
@@ -12,4 +12,8 @@ Make example for this keyboard (after setting up your build environment):
 
     make id80:default
 
+or for (CH-based) ISO layout:
+    
+    make id80:iso_ch
```
Comment (Erovia) [inline keyboards/id80/readme.md:17]:
```suggestion
Flashing example for this keyboard:

    make id80:default:flash
```

I don't think it's necessary to add an example for the ISO layout, as the process is fundamentally the same.
But if you are already polishing this board's support, I'd like to ask you to add the flashing example and clear instructions on how to reset the board into bootloader mode.

File: keyboards/id80/keymaps/iso_ch/readme.md [hunk 9]
```diff
@@ -0,0 +1 @@
+# The default keymap for id80
```
Comment (Erovia) [inline keyboards/id80/keymaps/iso_ch/readme.md:1]:
```suggestion
# The default ISO keymap for id80
```

File: keyboards/id80/info.json [hunk 10]
```diff
@@ -6,8 +6,10 @@
...
+        },
+        "LAYOUT_iso": {
```
Comment (Erovia) [inline keyboards/id80/info.json:11]: image

I think the enter and the backslash needs a bit love.
Not an expert on `info.json` formatting, but using the `ks` key for the ISO enter might be helful.

**Review by Erovia — CHANGES_REQUESTED** [review]

File: keyboards/id80/info.json (hunk 10 above)
Comment (phi-0) [reply keyboards/id80/info.json:11]: alright, I will have another look and make adjustments

**Pull request closed without merging**